javascript/javascript2/week3/warmupp/warmup.js

```diff
@@ -0,0 +1,104 @@
+let button=document.querySelector('button')
+let locationButton=document.querySelector('#location')
+//1  
+let loadTimer=()=>setTimeout(() => {
+ console.log("Called after 2.5 seconds");   
+}, 2500);
+
+document.addEventListener("DOMContentLoaded", loadTimer);
+
+//2
+function logStringDelay(delay,stringToLog){
+    setTimeout(() => {
+     console.log(stringToLog);   
+    }, 1000*delay);
+}
+
+logStringDelay(6,'This string logged after 6 seconds')
+logStringDelay(2,'This string logged after 2 seconds')
+
+//3
+
+let logDelay=()=>logStringDelay(5,'This string logged after 5 seconds')
+button.addEventListener('click',logDelay)
+
+
+//4
+const earthLogger=()=>console.log('Earth');
+const saturmLogger=()=>console.log('Satrum');
+function planetLogFunction(callBack){
+    callBack()
+}
+planetLogFunction(earthLogger);
+planetLogFunction(saturmLogger);
+
+
+//5
+
+  function success(pos) {
+    var crd = pos.coords;
+  
+    console.log(`This is the latitude : ${crd.latitude}`);
+    console.log(`This is the longitude: ${crd.longitude}`);
+    
+  }
+  
+  
+  
+locationButton.addEventListener('click',()=>navigator.geolocation.getCurrentPosition(success)
+)
+
+  function runAfterDelay(delay,callback){
+setTimeout(() => {
+    callback()
+}, delay*1000);
+  }
+
+  runAfterDelay(4,function(){
+      console.log('should be logged after 4 seconds');
+  })
+  runAfterDelay(3,()=> console.log('should be logged after 3 seconds'))
+  runAfterDelay(5,()=> console.log('should be logged after 5 seconds'))
+
+  //8
+  let doubleClick=0
+document.addEventListener('click',()=>{
+       doubleClick++
+      setTimeout(() => {
+        if(doubleClick==2){
+            console.log('double Click');
+        }
+      }, 500);
+  })
+  

```

Comment:
Thats nice, there is another event for double click in js(just for info) https://developer.mozilla.org/en-US/docs/Web/API/Element/dblclick_event